components/servo/tests/site_data_manager.rs

```diff
@@ -0,0 +1,207 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+mod common;
+
+use std::rc::Rc;
+
+use http_body_util::combinators::BoxBody;
+use hyper::body::{Bytes, Incoming};
+use hyper::{Request as HyperRequest, Response as HyperResponse};
+use net::test_util::{make_body, make_server};
+use servo::{JSValue, SiteData, StorageTypes, WebViewBuilder};
+
+use crate::common::{ServoTest, WebViewDelegateImpl, evaluate_javascript};
+
+fn sites_equal_unordered(actual: &[SiteData], expected: &[SiteData]) -> bool {
+    let mut actual = actual.to_vec();
+    let mut expected = expected.to_vec();
+
+    actual.sort_by(|a, b| a.name().cmp(&b.name()));
+    expected.sort_by(|a, b| a.name().cmp(&b.name()));
+
+    actual == expected
+}
+
+macro_rules! site_data_vec {
+    ( $( ($url:expr, $storage:expr) ),+ $(,)? ) => {
+        &[
+            $(
+                SiteData::new(
+                    $url.origin().ascii_serialization(),
+                    $storage,
+                ),
+            )+
+        ]
+    };
+}
+
```

Comment:
Would you be willing to turn this macro a function to turn a vector of tuples into structs? Performance isn't critical here and macros make it difficult to follow the code in the IDE.